### PR TITLE
DCOS_OSS-1564: Let's update webpack (2nd try)

### PR DESCRIPTION
**Pull request by juliangieseke**
Something went wront when we tried to rebase the old PR (#2699), this was the latest backup and work is continued here 👍 

please refer to the old PR for more information.

### CODE REVIEW

File: webpack/webpack.config.babel.js [hunk 1]
```diff
@@ -1,176 +0,0 @@
...
-      }
-    ],
-    postLoaders: [
-      {
-        test: /\.html$/,
-        exclude: /node_modules/,
-        loader: StringReplacePlugin.replace({
-          replacements: [
-            {
-              pattern: /<!--\[if\sBOOTSTRAP-CSS\]><!\[endif\]-->/g,
```
Comment (jaxesn) [inline webpack/webpack.config.babel.js]: Is this stuff happening somewhere else now, or is one of the webpack plugins taking care of it?

File: src/index.html [hunk 2]
```diff
@@ -1,43 +1,44 @@
...
+  <link rel="apple-touch-icon" sizes="60x60" href="./img/components/icons/favicon-apple-60x60.png">
+  <link rel="apple-touch-icon" sizes="76x76" href="./img/components/icons/favicon-apple-76x76.png">
+  <link rel="apple-touch-icon" sizes="120x120" href="./img/components/icons/favicon-apple-120x120.png">
+  <link rel="apple-touch-icon" sizes="152x152" href="./img/components/icons/favicon-apple-152x152.png">
+  <link rel="apple-touch-icon-precomposed" sizes="60x60" href="./img/components/icons/favicon-apple-60x60.png">
+  <link rel="apple-touch-icon-precomposed" sizes="76x76" href="./img/components/icons/favicon-apple-76x76.png">
+  <link rel="apple-touch-icon-precomposed" sizes="120x120" href="./img/components/icons/favicon-apple-120x120.png">
+  <link rel="apple-touch-icon-precomposed" sizes="152x152" href="./img/components/icons/favicon-apple-152x152.png">
+  <script type="text/javascript">(function () { var PAGE_LOADED_TIME = Date.now(); window.getPageLoadedTime = function () { return PAGE_LOADED_TIME; } })();</script>
+  <!--[if BOOTSTRAP-CSS]><![endif]-->
```
Comment (jaxesn) [inline src/index.html]: Related to my other question, is this block still needed?

Comment (DanielMSchmidt) [reply src/index.html]: I have searched for bootstrap and did not find it, that is why I removed the StringReplacePlugin, might be that I was mistaken 🤔 But if we don't need it we can remove it here, too

**Comment by ahoskins** [comment]
@juliangieseke It's really unideal to close a PR because of a failed rebase. All the history of comments gets kind of lost and forgotten. I encourage using `git reflog` next time to undo any damage, and continue on the same PR!

**Comment by juliangieseke** [comment]
@ahoskins you're right, but if you check out the branch from the old PR, it is in a different state, then the PR here on the website. we somehow managed to break Github. I'm not exactly sure what happened, but after we recovered the "before-rebase-state" from Daniels machine into this new branch, we tried to force-push it against the old one. It failed and from that point we werent able to push to that branch at all 🙈 might be us doing something wrong, but since jackson already commented on this PR I also didnt want to lose his comments - so we sticked to this one.

**Comment by juliangieseke** [comment]
rebased this branch 👍

**Comment by DanielMSchmidt** [comment]
@juliangieseke As far as I know you are still working on this, right? I added the label, please remove it if I am wrong 🙈

File: webpack/webpack.config.babel.js (hunk 1 above)
Comment (juliangieseke) [reply webpack/webpack.config.babel.js]: we have implemented it here: 👍

Comment (juliangieseke) [reply webpack/webpack.config.babel.js]: here: https://github.com/dcos/dcos-ui/pull/2817/files#diff-307854f18e096bcb2404b859d732dcdc

File: scripts/inline-bootstrap-css.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env node
```
Comment (jaxesn) [inline scripts/inline-bootstrap-css.js]: This is so much more clear, comment def helps too. 👍

Comment (DanielMSchmidt) [reply scripts/inline-bootstrap-css.js]: It's not nice that we have to do it, but this is at least the most readable way possible 🙈

File: scripts/ci/run-integration-tests [hunk 4]
```diff
@@ -44,10 +44,11 @@ function runTestFile {
...
 }
 
+find $PROJECT_ROOT/tests -type f -name '*-cy.js' | while read file; do runTestFile "$file"; done
 
-for f in $PROJECT_ROOT/tests/**/*-cy.js; do
-  runTestFile "$f"
-done
+# for f in $PROJECT_ROOT/tests/**/*-cy.js; do
+#   runTestFile "$f"
+# done
```
Comment (Fabs) [inline scripts/ci/run-integration-tests]: shouldn't it be removed instead of commented out? Same for other instances on this file.

Comment (juliangieseke) [reply scripts/ci/run-integration-tests]: oh, yes. good catch! will fix it tomorrow when resolving the conflicts.

File: webpack/proxy.template.js [hunk 5]
```diff
@@ -22,5 +22,8 @@ module.exports = {
   //     req.url = req.url.replace(/^\/mesos/, '');
   //   }
   // },
-  "*": "http://m1.dcos"
+  "*": {
```
[the remark this answers is not in the document]
Comment (juliangieseke) [reply webpack/proxy.template.js]: it allows us to proxy https with an invalid cert: https://webpack.js.org/configuration/dev-server/#devserver-proxy

**Comment by juliangieseke** [comment]
should be good to go now 👍 cant find anything which is broken anymore 💃

**Comment by nLight** [comment]
Conflicts don't let the CI run

**Comment by DanielMSchmidt** [comment]
Oh my god, not this again 😢

**Comment by juliangieseke** [comment]
Are there information missing to properly review this one?

**Comment by nLight** [comment]
I'll do it tomorrow in the morning. Line by line

review-2

**Comment by Poltergeist** [comment]
@juliangieseke Sorry if you explained it already somewhere, but why are there changes in the less files and the html files here. For example changing from a file url to a data url as a background image?

**Comment by juliangieseke** [comment]
the old implementation was a workaround which was needed for our old webpack 1 setup, when we changed the way the bootstrap part is generated we needed to test it (by changing it back to the correct implementation) and I left it in because it was changed already and it was a (very bad) workaround for a (also bad) workaround in our old setup. 👍

**Comment by juliangieseke** [comment]
oh, and its the other way round: from data to file 👍 (data was the workaround)

**Comment by Fabs** [comment]
I also ran:
`npm run`
`npm test`
`npm build`
`npm run testing`

File: webpack/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+      "node_modules"
+    ]
+  },
+  node: {
+    fs: "empty" // Jison loader fails otherwise
+  },
+  devServer: {
+    // TODO: https://webpack.js.org/configuration/dev-server/#devserver-hot
+    contentBase: path.join(__dirname, "../dist"),
+    open: true,
```
Comment (ahoskins) [inline webpack/webpack.config.js]: Why? This is kind of annoying imo.

**Pull request merged**